### ISSUE TITLE
VGG19 con validación cruzada

### ISSUE THREAD

**Issue by NuriaVP** [NuriaVP/Transfer-Learning-EDM#27]
Analizar cómo añadir en el código que llevamos hasta el momento la validación cruzada para mejorar los resultados.

**Comment by NuriaVP** [comment]
**Información general sobre el concepto de validación cruzada**

https://www.aprendemachinelearning.com/sets-de-entrenamiento-test-validacion-cruzada/

**Comment by NuriaVP** [comment]
**Otra fuente de información interesante:**

https://pythonguia.com/scikit-aprende-validacion-cruzada-guia-util/

**Comment by NuriaVP** [comment]
Vamos a hacer un entrenamiento de VGG19 (dado que parece que da mejores resultados que VGG16) empleando validación cruzada. Lo haremos con los datos que llevan preprocesamiento gaussiano y usando class_weight_compute. Pero no utilizaremos data augmentation porque aún no funciona bien.

**Comment by NuriaVP** [comment]
Intento fallido validación cruzada en el commit: 919f42374e7131be251d8d747fc86b5cc9cb278e

En esta tarea he utilizado VGG19 para evaluar el entrenamiento con cross_val_score, pero me ha dado un error que desconozco de dónde ha podido salir.

**Comment by NuriaVP** [comment]
**Nuevas fuentes de información**

[https://scikit-learn.org/stable/modules/cross_validation.html](url)

[https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html](url)

[https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html](url)

**Comment by NuriaVP** [comment]
Tarea que funciona con VGG19 y cross validation: e5babf43a37b88fc1fbe576f22889fedee1a1303

En esta tarea tenemos el código para probar cross_val_score() y cross_val_predict() con el modelo VGG19 y el conjunto de datos OCT / iPhone / Samsung divididos con train_test_split. Funciona bien pero aún se pueden perfeccionar cosas, al final aparecen las métricas de evaluación de cómo de buena es la predicción empleando este procedimiento.

**Comment by NuriaVP** [comment]
**Resultados de la prueba del commit anterior**

**cross_val_score** 
[0.73770492 0.73770492 0.73770492 0.68852459 0.73770492]
0.73 accuracy with a standard deviation of 0.02

**cross_val_predict**
- AUC: 0.4920833333333334
- F1: 0.6267840538797255
- Accuracy: 0.7081967213114754
- Tabla de contingencia: 
[[213  12]
 [ 77   3]]

**Comment by NuriaVP** [comment]
**Foro donde se explica como equilibrar los grupos con el parámetro fit_params de cross_val_score**

https://stackoverflow.com/questions/35545733/how-do-you-use-fit-params-for-randomizedsearch-with-votingclassifier-in-sklearn

**Comment by NuriaVP** [comment]
Tarea realizada en el commit: aefd676de2c24ee79a564e7f69be93c861ac4394

He hecho algunos cambios en el cuaderno "VGG19 cross validation" para parametrizar las pruebas.
